pkgs/applications/backup/timeshift/unwrapped.nix

```diff
@@ -26,17 +26,20 @@ stdenv.mkDerivation rec {
     hash = "sha256-2qNLgUZLMcfoemdLvvjdkR7Ln5loSKGqbd402y5Id8k=";
   };
 
-  patches = [
-    ./timeshift-launcher.patch
-  ];
-
   postPatch = ''
-    while IFS="" read -r -d $'\0' FILE; do
+    for FILE in src/Core/Main.vala src/Utility/Device.vala; do
       substituteInPlace "$FILE" \
-        --replace "/sbin/blkid" "${util-linux}/bin/blkid"
-    done < <(find ./src -mindepth 1 -name "*.vala" -type f -print0)
+        --replace-fail "/sbin/blkid" "${lib.getExe' util-linux "blkid"}"
+    done
+
     substituteInPlace ./src/Utility/IconManager.vala \
-      --replace "/usr/share" "$out/share"
+      --replace-fail "/usr/share" "$out/share"
+
+    # Substitute app_command to look for the `timeshift-gtk` in the same directory as the executed `timeshift-launcher`.
+    # Substitute the `pkexec ...` as a hack to run a GUI application like Timeshift as root without setting up the corresponding pkexec policy.
+    substituteInPlace ./src/timeshift-launcher \
+      --replace-fail "app_command='timeshift-gtk'" ${lib.escapeShellArg ''app_command="$(realpath "$(dirname "$0")")/timeshift-gtk"''} \
```

Comment:
Can we utilize $out here instead?

Reply:
Then, this should be done in the `wrapper.nix` instead.

Reply:
I added a commit to implement this suggestion.